client/db/bolt/db.go

```diff
@@ -2075,6 +2076,40 @@ func (db *BoltDB) DeleteInactiveMatches(ctx context.Context, olderThan *time.Tim
 	return nil
 }
 
+// SaveDisabledRateFetchers updates disabled exchange rate fetchers.
+func (db *BoltDB) SaveDisabledRateFetchers(disabledFetchers []string) error {
+	return db.Update(func(tx *bbolt.Tx) error {
+		bkt := tx.Bucket(appBucket)
+		if bkt == nil {
+			return fmt.Errorf("failed to open %s bucket", string(appBucket))
+		}
+		disabledByte := []byte(strings.Join(disabledFetchers, ","))
+		return bkt.Put(disabledRateFetchersKey, disabledByte)
+	})
+}
+
+// DisabledRatefetchers retreives a map of disabled exchange rate fetchers.
+func (db *BoltDB) DisabledRatefetchers() (disabledFetchers []string, err error) {
+	return disabledFetchers, db.View(func(tx *bbolt.Tx) error {
+		bkt := tx.Bucket(appBucket)
+		if bkt == nil {
+			return fmt.Errorf("no %s bucket", string(appBucket))
+		}
+		disabledByte := bkt.Get(disabledRateFetchersKey)
+		disabledString := string(disabledByte)
+		if disabledString == "" {
+			return nil
+		}
+		disabled := strings.Split(disabledString, ",")
+		for _, token := range disabled {
+			if token != "" {
+				disabledFetchers = append(disabledFetchers, token)
```

Comment:
Could you create this slice before the loop? `disabledFetchers = make([]string, len(disabled))`